### ISSUE TITLE
Revisit size and reported min/max of NLDAS meteo files

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/lake-temp-lstm-static-data-release#45]
From Sam: 

> - What is the 0.379366 in the NLDAS file names?  
> - I expected each of these .csv files to be the same length – basically one row for every date over the entire period. However, the files do not have the same length. The best I can tell is that the data is duplicated for however many lakes are crosswalked to that cell – e.g., if there are 5 lakes in that cell, there will be (n dates X 5) rows in that file. There are random NA values in rows when these files are read in, too.  
> - How did you generate min and max in the metadata? They do not match min/max values in the files. For example, in file NLDAS_time[0.379366]_x[194]_y[181].csv, minimum air temperature is -36.6 but the reported min in the metadata is -27.2. Several other examples.

**Comment by lindsayplatt** [comment]
> What is the 0.379366 in the NLDAS file names?

In the full filepath: ``. It is the same value for every file, which means we can probably just remove it. From what I can tell, it was used as a unique identifier for the time range represented (see construction of [filepath here](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/6_drivers/src/nldas_feather_utils.R#L86), and [the specified time range here](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/6_drivers_fetch.yml#L25)). It is likely leftover from when there were different files with different time ranges. This is saying that the file starts at timestep of `0` and ends at timestep of `379366` (where timesteps are hours since the starting date-time).

Solution is to remove this field from the filepaths since they all have the same value and we have other ways to report time in the metadata.

**Comment by lindsayplatt** [comment]
> I expected each of these .csv files to be the same length – basically one row for every date over the entire period.

Me too! Upon further investigation, it looks as if we have a lot of duplicated info in these files. When you de-duplicate, they all have the same length. Must be something with how we were using them in `lake-temperature-process-models`? I copied the CSVs from `lake-temp/lake-temperature-process-models/1_prep/in/NLDAS_GLM_csvs`. Maybe those were duplicated purposefully in prep for modeling?

```r
library(tidyverse)
x1 <- read_csv("tmp_data/NLDAS/NLDAS_time[0.379366]_x[227]_y[164].csv")
x2 <- read_csv("tmp_data/NLDAS/NLDAS_time[0.379366]_x[229]_y[174].csv")
nrow(x1)
nrow(x2)
nrow(distinct(x1))
nrow(distinct(x2))

> nrow(x1)
[1] 900998
> nrow(x2)
[1] 284525

> nrow(distinct(x1))
[1] 15807
> nrow(distinct(x2))
[1] 15807
```

My solution for now will be to add a step that loads and uses `distinct()` for each of these files ...

**Comment by lindsayplatt** [comment]
Now each of the zips are under 1 GB. I am going to drop the number of zips down to 2 (which will mean that the `lake_metadata.csv` file will need a rebuild).

**Comment by lindsayplatt** [comment]
To revise min/max, I did this:

```r
# Calc min/max of each column:
all_nldas_meteo_csvs <- list.files("tmp_data/NLDAS", full.names = TRUE)
nldas_meteo <- purrr::map(all_nldas_meteo_csvs, function(fn) {
  data.table::fread(fn, drop = "time")[, c(min = lapply(.SD, min),
                                           max = lapply(.SD, max))]
}) %>% data.table::rbindlist()

# Min values
nldas_meteo_min <- nldas_meteo %>% as_tibble() %>% 
  select(starts_with("min.")) %>% 
  pivot_longer(cols = everything(), values_to = 'min_vals', names_to = "variable") %>% 
  mutate(variable = gsub('min.', '', variable)) %>% 
  group_by(variable) %>% summarize(min = min(min_vals))

# Max values
nldas_meteo_max <- nldas_meteo %>% as_tibble() %>% 
  select(starts_with("max.")) %>% 
  pivot_longer(cols = everything(), values_to = 'max_vals', names_to = "variable") %>% 
  mutate(variable = gsub('max.', '', variable)) %>% 
  group_by(variable) %>% summarize(max = max(max_vals))

left_join(nldas_meteo_min, nldas_meteo_max)

# DO TIME THIS TIME!
nldas_meteo_tm <- purrr::map(all_nldas_meteo_csvs, function(fn) {
  data.table::fread(fn, select = "time")[, c(min = lapply(.SD, min),
                                             max = lapply(.SD, max))]
}) %>% data.table::rbindlist()
min(nldas_meteo_tm$min.time)
max(nldas_meteo_tm$max.time)
```

**Comment by lindsayplatt** [comment]
Closed by #78
